### PR TITLE
Add Workspace and Connection Geography Support to API

### PR DESCRIPTION
**Pull request by pmossman**
## What
Addresses https://github.com/airbytehq/airbyte-cloud/issues/2278

## How
- Adds new endpoint `v1/web_backend/geographies/list` which returns available Geographies
  - Will be just `['auto']` in OSS, and `['auto', 'us', 'eu']` in Cloud
- Modifies `/v1/workspaces/update` to support a new property called `defaultGeography`
  - This endpoint is already PATCH-style, which means:
    - If this property is present on the update request, will set the workspace's `defaultGeography` column to the value
    - If this property is not present on the update request, will leave the workspace's current `defaultGeography` unmodified
- Modifies the `/v1/connections/create` and `/v1/web_backend/connections/create` endpoint to support a new optional property called `geography`
  - If present on the create request, will set the Connection's geography column to the value
  - If not present, the Connection will use whatever is set on its workspace's `defaultGeography`.
- Modifies the `/v1/connections/update` and `/v1/web_backend/connections/update` endpoint to support a new optional property called `geography`
  - This endpoint is already PATCH-style, which means:
     - If this property is present, will set the Connection's 'geography' column to the value
     - If not present, will leave the Connection's 'geography' unmodified
  

## Recommended reading order
1. Start with the various `.yaml` files that define the Geography type and various schema updates
2. WebBackendGeographiesHandler.java
3. ConnectionsHandler.java
4. WorkspacesHandler.java
5. Other non-test changes, all relatively minor
6. Tests

### CODE REVIEW

File: airbyte-bootloader/src/main/java/io/airbyte/bootloader/BootloaderApp.java
Comment (davinchia) [inline airbyte-bootloader/src/main/java/io/airbyte/bootloader/BootloaderApp.java:291]: nice touch

File: airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendGeographiesHandler.java
Comment (davinchia) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendGeographiesHandler.java:19]: can we add more detail on what this means for OSS? e.g. a single task queue

File: airbyte-api/src/main/openapi/config.yaml
Comment (terencecho) [inline airbyte-api/src/main/openapi/config.yaml:2043]: Should we also mention that we may return `auto`?

We can also add that users can expect the country codes to follow this iso standard: https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2 (minus `auto`), since that's what the frontend will be expecting so they can easily link them to country flags.

File: airbyte-api/src/main/openapi/config.yaml
Comment (terencecho) [inline airbyte-api/src/main/openapi/config.yaml:2467]: lowercase vs capitalized, does it matter here? i think the database enums are capitalized. We should double check if the frontend has a preference on how it's returned to them

**Review by terencecho — APPROVED** [review]
lgtm, left a couple of comments. I think it's probably worth waiting for someone on FE to give another approval before merging.

**Review by ambirdsall — APPROVED** [review]
The API contracts look reasonable to me. I like that it seems the frontend can use the same code path for OSS and cloud, and will simply fetch different geography lists based on context (hiding a geography selection UI if the only geography is `auto` is a reasonable piece of busines logic, and cleaner than hardcoding distinct OSS/cloud behaviors); easy to work with and extend in various directions as the product evolves.

My only issue is fetching data with a `POST` verb, but that's a sufficiently well-established convention that it wouldn't be appropriate to add in an ad hoc one-off `GET` endpoint.

File: airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendGeographiesHandler.java
Comment (davinchia) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendGeographiesHandler.java:29]: remind me: why/how does the Cloud api call this?

Comment (davinchia) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendGeographiesHandler.java:29]: Is the idea to explicitly call this method in the wrapped server in Cloud?

**Review by davinchia — APPROVED** [review]
Looks sane to me!

One question to make sure I understand how we want to call this in Cloud.

File: airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendGeographiesHandler.java
Comment (pmossman) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendGeographiesHandler.java:29]: Yeah exactly, in the Cloud wrapped server, we'd call this specific endpoint instead of the OSS-specific endpoint. I think this is a short-term solution that we'll iterate on once we have more concrete needs around specialized OSS use-cases. I think it should serve our needs for now and should be pretty easy to adjust in the future

**Pull request merged**
